### PR TITLE
add documentation about samplers

### PR DESCRIPTION
**Pull request by chad-g-adams**
As discussed on Elixir slack this week, I've added a little documentation around custom samplers and how to create them. 

I only have elixir expertise (not erlang), so if you want the code samples to be shown in erlang also then I'll need you to tell me exactly what to write for it (or feel free to push to the branch yourself)

### CODE REVIEW

File: apps/opentelemetry/README.md [hunk 1]
```diff
@@ -63,6 +63,35 @@ Disabling the creation of Tracers at start-up means when the macros are used to
 start Spans the name and version of the Tracer creating those Spans will be
 undefined, but everything will continue to work as before.
 
+### Samplers
+
+Sampling is a mechanism to control the number of traces collected and sent to the backend. A Sampler is responsible for making a decision on whether to sample a trace or not. There are several [built-in samplers](https://opentelemetry.io/docs/reference/specification/trace/sdk/#built-in-samplers) available, and it's possible to create custom samplers.
+
+| OS                             | Application              | Default   | Type    |
+|:-------------------------------|:-------------------------|:----------|:--------|
+| OTEL_TRACES_SAMPLER             | sampler                 | parentbased_always_on | always_on, always_off, traceidratio, parentbased_always_on, parentbased_always_off, parentbased_traceidratio |
```
Comment (chad-g-adams) [inline apps/opentelemetry/README.md:72]: I grabbed these options from the OTel spec, but I did not verify if there were indeed all supported by this implementation or not - please confirm :)

Comment (tsloughter) [reply apps/opentelemetry/README.md:72]: Yes, they are all supported.

File: apps/opentelemetry/README.md [hunk 2]
```diff
@@ -63,6 +63,35 @@ Disabling the creation of Tracers at start-up means when the macros are used to
...
+
+#### Custom Sampler
+
+To create a custom sampler, implement a module with behaviour `otel_sampler`, and then install your custom sampler when configuring the `opentelemetry` application
+
+``` elixir
+defmodule MySampler do
+  @behaviour :otel_sampler
+
+  ...
```
Comment (tsloughter) [inline apps/opentelemetry/README.md:84]: Could you add examples of the 3 functions that are part of the behaviour?

Comment (chad-g-adams) [reply apps/opentelemetry/README.md]: Done. Let me know what you think!

**Review by tsloughter — APPROVED** [review]

**Pull request merged**